tests/extract/conftest.py

```diff
@@ -62,9 +62,42 @@ def create_client_config():
     return config
 
 
+@pytest.fixture
+def create_bad_client_config():
+    """create invalid s3 configuration dictionary with bad endpoint_url value
+    to be passed to boto3.client("s3", **s3_config)"""
+    config = {}
+    load_dotenv()
+
+    config["aws_access_key_id"] = os.getenv("S3FILESTORE__AWS_ACCESS_KEY_ID")
+    config["aws_secret_access_key"] = os.getenv("S3FILESTORE__AWS_SECRET_ACCESS_KEY")
+    config["region_name"] = os.getenv("S3FILESTORE__REGION_NAME")
+    config["endpoint_url"] = "garbage"
+
+    return config
+
+
 @pytest.fixture
 def create_client(create_client_config):
     """create a boto3.client
     create_client_config (dict)     :   configuration file.
     """
-    return create_s3_client(create_client_config)
+    S3_client = create_s3_client(create_client_config)
+    yield S3_client
```

Comment:
is `yield` not effectively the same as a return in python? in JS it's to return a generator and all lines following will be ignored

Reply:
no, they're not the same. lines proceeding `yield` are executed. this provides an opportunity for cleanup to occur after the fixture object has served its purpose.

Reply:
read the docs and now i get what you were doing. nice!